Web_Development/exercise-05/ex5.js

```diff
@@ -0,0 +1,162 @@
+
+  let pokemonWeights = document.createElement("h4");
+  pokemonWeights.innerText = `Weight: ${weight}`;
+
+  let pokemonBase_experience = document.createElement("h4");
+  pokemonBase_experience.innerText = `Base experience: ${base_experience}`;
+
+  let pokemonImg = document.createElement("img");
+  pokemonImg.id = "pokemonImg-hover";
+  pokemonImg.src = urlData.sprites.front_default;
+
+  let showPokemonDetails = document.createElement("button");
+  showPokemonDetails.innerText = "Show More Details";
+
+  showPokemonDetails.addEventListener("click", () => {
+    showMoreDetails.srcdoc = `<!DOCTYPE html><html>
+    <head>
+    <style>
+    #show_more__moves{
+      font-size: 20px;
+      color: black;
+      text-align: center;
+      }
+    #more_moves{
+      font-size: 10px;
+      padding: 10px;
+      background-color: black;
+      border-radius: 5px;
+      color: white;
+      cursor: pointer;
+    }
+    </style>
+
+    </head>
+    <body>
+    <img width="150" src="${
+      urlData.sprites.front_default
+    }"/><h2>Name: ${name}</h2> <h4>ID: ${id}</h4> <h4>Weight: ${weight}</h4> <h4>Base experience: ${base_experience}</h4> <h4> Type: ${
+      urlData.types[0].type.name
+    }</h4> <h4> Moves: ${urlData.moves[0].move.name}, ${
+      urlData.moves[1].move.name
+    }, ${urlData.moves[2].move.name}, ${urlData.moves[3].move.name}, ${
+      urlData.moves[4].move.name
+    }</h4>
+    <button onclick="pokemonMoves()" id="more_moves">Show More Moves</button>
+    <p id=show_more__moves></p>
+        <script>
+        function pokemonMoves(){
+          show_more__moves.innerHTML= "${urlData["moves"]
+            .map((element, index) => (index > 4 ? (element.move.name) + "," : ""))
+            .join(" ")}";
+        }
+        </script>
+    </body>
+    </html>`;
+  });
+
+  pokemonImg.addEventListener("mouseout", () => {
+    pokemonImg.src = urlData.sprites.front_default;
+  });
+  pokemonImg.addEventListener("mouseover", () => {
+    pokemonImg.src = urlData.sprites.back_default;
+  });
+  createPokemonContainer.append(
+    pokemonImg,
+    pokemonName,
+    pokemonId,
+    pokemonWeights,
+    pokemonBase_experience,
+    showPokemonDetails
+  );
+  pokemon.appendChild(createPokemonContainer);
+};
+
+const searchPokemonName = () => {
+  let searchPokemon = document.getElementById("search-pokemon").value;
+  let searchArray = URL_info.filter((element) =>
+    element.name.includes(searchPokemon)
+  );
+  if (pokemon.innerHTML == "") {
+    pokemon.innerHTML = `<h1>No results found</h1>`;
+  } else {
+    pokemon.innerHTML = "";
+    searchArray.forEach((element) => {
+      fetchPokemonUrlDetails(element);
+    });
+  }
+};
+
+const sortPokemon = (value) => {
+  let result = Array.from(
+    document.getElementById("pokemon-display").children
+  ).sort((x, y) => {
+    x = String(x.children[value].innerText).split(":")[1];
+    y = String(y.children[value].innerText).split(":")[1];
+    return x - y;
```

Comment:
same variable names as parameter names are being reused for the local variables which is not a good practice
also, since we have only one usage of local variables, it can be omitted and and the arrow function shorthand could be used.